test/utils.js

```diff
@@ -46,6 +46,23 @@ async function isUserInGroup(groupUser, group) {
   return userPresent;
 }
 
+async function waitTillUserInGroup(user, group, condition) {
+  let userInGroup = await isUserInGroup(user, group);
+  let timeOut = 0;
+  while (userInGroup !== condition) {
+    userInGroup = await isUserInGroup(user, group);
+    if (userInGroup === condition) {
+      return userInGroup;
+    }
+
+    await delay(1000);
+    timeOut++;
+    if (timeOut === 30) {
```

Comment:
If I understand this correctly, we could have a test take 30 seconds to fail/succeed?